### PR TITLE
Added online evaluation RAG using Langfuse

### PR DESCRIPTION
**Pull request by visitani**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: knowledge-bases/6_customized-rag-retreive-api-langchain-claude-v2-online-evaluation-ragas.ipynb [hunk 1]
```diff
@@ -0,0 +1,1075 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Building and evaluating Q&A Application using Knowledge Bases for Amazon Bedrock using Online Evaluation of RAG with Langfuse & Ragas framework"
```
[the remark this answers is not in the document]
Comment (visitani) [reply knowledge-bases/6_customized-rag-retreive-api-langchain-claude-v2-online-evaluation-ragas.ipynb:7]: updated

File: knowledge-bases/6_customized-rag-retreive-api-langchain-claude-v2-online-evaluation-ragas.ipynb [hunk 2]
```diff
@@ -0,0 +1,1075 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Context\n",
+    "\n",
+    "In this notebook, we will dive deep into building Q&A application using Retrieve API provide by Knowledge Bases for Amazon Bedrock, along with LangChain, Langfuse and RAGAS for evaluating the responses. Here, we will query the knowledge base to get the desired number of document chunks based on similarity search, prompt the query using Anthropic Claude, and then we Collect & calculate scores for your LLM responses effectively using metrics, such as faithfulness, answer relevancy, context precision, harmfulness based expectations and then ingest traces to Langfuse.\n",
```
[the remark this answers is not in the document]
Comment (visitani) [reply knowledge-bases/6_customized-rag-retreive-api-langchain-claude-v2-online-evaluation-ragas.ipynb:16]: corrected wording

File: knowledge-bases/6_customized-rag-retreive-api-langchain-claude-v2-online-evaluation-ragas.ipynb [hunk 3]
```diff
@@ -0,0 +1,1075 @@
...
+    "1. Load the documents into the knowledge base by connecting your s3 bucket (data source). \n",
+    "2. Ingestion - Knowledge base will split them into smaller chunks (based on the strategy selected), generate embeddings and store it in the associated vectore store.\n",
+    "\n",
+    "![data_ingestion.png](./images/data_ingestion.png)\n",
+    "\n",
+    "3. API Keys - You require a Langfuse public and secret key to get started. follow the below steps and find them in your project settings.\n",
+    "\n",
+    "    1. Create [Langfuse account](https://cloud.langfuse.com/auth/sign-up)\n",
+    "    2. Create a new project\n",
+    "    3. Create new API credentials in the project settings, On the Settings page, click Create new API keys to generate a new secret and public key pair. Store these keys securely - you'll need them in this notebook.\n",
```
[the remark this answers is not in the document]
Comment (visitani) [reply knowledge-bases/6_customized-rag-retreive-api-langchain-claude-v2-online-evaluation-ragas.ipynb:42]: updated

File: knowledge-bases/README.md [hunk 4]
```diff
@@ -11,6 +11,22 @@
...
 
+- [5_customized-rag-retreive-api-langchain-claude-v2-evaluation-ragas.ipynb](./5_customized-rag-retreive-api-langchain-claude-v2-evaluation-ragas.ipynb) - If you are interested in building Q&A application using Retrieve API provide by Knowledge Bases for Amazon Bedrock, along with LangChain and RAGAS for evaluating the responses, try this sample.
+
+- [6_customized-rag-retreive-api-langchain-claude-v2-online-evaluation-ragas.ipynb](./6_customized-rag-retreive-api-langchain-claude-v2-online-evaluation-ragas.ipynb) - The popularity of large language models (LLMs) is skyrocketing, and with it the    need to observe and analyze their performance. Tracing model usage in production and getting detailed insights on quality, cost and speed are crucial for the continued growth of generative AI apps.  Now let's explore Langfuse, another emerging project tackling observability for these complex systems.
```
[the remark this answers is not in the document]
Comment (visitani) [reply knowledge-bases/README.md:16]: added extra space for allignment.

File: knowledge-bases/README.md [hunk 5]
```diff
@@ -11,6 +11,22 @@
...
+
+    Langfuse aims to provide granular visibility into model invocation traces and metrics like accuracy, latency and cost per query. With advanced analytics and visualization, it can help teams optimize performance, reduce expenses and identify issues early. As generative AI enters the mainstream, Langfuse and similar tools will be key enablers for delivering reliable, cost-effective services at scale. 
+
+    In this notebook, We will use Ragas to run the evaluations for each trace item and score them. This gives you better idea of how each call to RAG pipelines is performing. You compute the score with each request from getting question from the user and fetch context from the Knoweldge base then pass the question and the contexts to the LLM to generate the answer. All these step are logged as spans in a single trace in langfuse. You can read more about traces and spans from the [langfuse documentation](https://langfuse.com/docs/tracing/overview).
```
[the remark this answers is not in the document]
Comment (visitani) [reply knowledge-bases/README.md:20]: updated

**Review by mani-aiml — APPROVED** [review]

**Pull request merged**